xbmc/interfaces/json-rpc/schema/types.json

```diff
@@ -460,14 +462,37 @@
   "Audio.Contributors": {
     "type": "array",
     "items": { "type": "object",
+      "description": "The artist and the role they contribute to a song",
       "properties": {
         "name": { "type": "string", "required": true },
         "role": { "type": "string", "required": true },
-        "artistid": { "type": "integer", "required": true }
+        "roleid": { "$ref": "Library.Id", "required": true },
+        "artistid": { "$ref": "Library.Id", "required": true }
       },
       "additionalProperties": false
     }
   },
+  "Audio.Artist.Roles": {
+    "type": "array",
+    "items": { "type": "object",
+      "description": "The various roles contributed by an artist to one or more songs",
+      "properties": {
+        "roleid": { "$ref": "Library.Id", "required": true },
+        "role": { "type": "string", "required": true }
+      },
+      "additionalProperties": false
+    }
+  },
+  "Audio.Artist.SongGenres": {
+    "type": "array",
+    "items": { "type": "object",
+      "description": "The various genres of one or more songs by an artist",
+      "properties": {
+        "songgenreid": { "$ref": "Library.Id", "required": true },
```

Comment:
I'm not sure if we should repeat the `songgenre` part here. The property is already called `songgenres` so repeating `songgenre` in the properties inside that property feels redundant. And if we name the properties `id` and `genre` (or `name`) they can be reused for other genre properties.